### PR TITLE
Posting to multiple Platform Done

### PR DESCRIPTION
**Pull request by hacker7599**

### CODE REVIEW

**Review by hacker7599 — COMMENTED** [review]
Reviewed The Changes Asking you to Review the code and merge it

File: app/Http/Controllers/Api/Ayrshare/AyrshareController.php [hunk 1]
```diff
@@ -115,18 +120,16 @@ public function ayrSocialMediaPosts(AyrSocialMediaPosts $request)
...
-                [
-                    'post' => $request->caption,
-                    'platforms' => $request->platforms,
-                    'mediaURLs' => $request->mediaURLs ?? [''],
-                    'instagramOptions' => $request->instagramOptions ?? [''],
-                ]
+                $request->all(),
+                $profile_key
             );
+            return dd($response->body());
```
Comment (jaydeepukani) [inline app/Http/Controllers/Api/Ayrshare/AyrshareController.php:132]: Remove dd from here

File: app/Http/Controllers/Api/Ayrshare/AyrshareController.php [hunk 2]
```diff
@@ -153,11 +156,20 @@ public function ayrForward(Request $request, $profileKey)
...
-        return json_decode(Http::withHeaders([
-            'Authorization' => 'Bearer '.$profile_key,
-        ])->withOptions(['verify' => true])->get(config('ayrshare.AYR_PROFILE_ENDPOINT'), []))->activeSocialAccounts;
+        $response = json_decode(Http::withHeaders([
+            'Authorization' => 'Bearer '.$request->profile_key,
+        ])->withOptions(['verify' => true])->get(config('ayrshare.AYR_PROFILE_ENDPOINT'), []));
+
+        if (property_exists($response, 'activeSocialAccounts')) {
+            return $response->activeSocialAccounts;
+        }else {
```
Comment (jaydeepukani) [inline app/Http/Controllers/Api/Ayrshare/AyrshareController.php:167]: Add space before else

**Review by jaydeepukani — CHANGES_REQUESTED** [review]
Please do CR changes

File: database/migrations/2022_01_05_100650_create_ayr_user_profiles_table.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('ayr_user_profiles', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('user_id');
+            $table->string('title');
+            $table->string('ref_id');
+            $table->string('profile_key');
```
Comment (jaydeepukani) [inline database/migrations/2022_01_05_100650_create_ayr_user_profiles_table.php:21]: Add unique key on profile_key

File: database/migrations/2022_01_05_100650_create_ayr_user_profiles_table.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            $table->string('authorized_on');
```
Comment (jaydeepukani) [inline database/migrations/2022_01_05_100650_create_ayr_user_profiles_table.php:22]: Use DateTime or timestamp field

File: app/Http/Controllers/User/PostScheduling/PostSchedulingController.php [hunk 5]
```diff
@@ -34,8 +36,11 @@ function index(Request $request)
 
     public function uploadPostMedia(UploadPostMediaRequest $request)
     {
-        // dd($request->all(), explode(',', $request->tags));
-        $tags = implode(",", preg_filter('/^/', '#', explode(',', $request->tags)));
+        if ($request->tags != "") {
+            $tags = implode(" ", preg_filter('/^/', '#', explode(',', $request->tags)));
+        }else {
+            $tags="";
```
Comment (jaydeepukani) [inline app/Http/Controllers/User/PostScheduling/PostSchedulingController.php:42]: Remove else and move this variable to before if

**Review by jaydeepukani — APPROVED** [review]
Do suggested changes on next PR.

File: database/migrations/2022_01_05_100650_create_ayr_user_profiles_table.php (hunk 4 above)
Comment (hacker7599) [reply database/migrations/2022_01_05_100650_create_ayr_user_profiles_table.php:22]: Changes Done will it work with -> Carbon::now()->toDateTimeString()

File: app/Http/Controllers/User/PostScheduling/PostSchedulingController.php (hunk 5 above)
Comment (hacker7599) [reply app/Http/Controllers/User/PostScheduling/PostSchedulingController.php:42]: Changes done

**Comment by hacker7599** [comment]
Thank You

**Pull request merged**